File: ruby/polystring.rb

class Polystring
	
	@@identifier = "en_US"
	
	def Polystring.identifier= (identifier)
		@@identifier = identifier
	end
	
	def Polystring.identifier
		return @@identifier
	end
	
	def initialize(raw)
		@raw = raw
	end
	
	def to_s
		locale
	end
	
	def raw
		@raw
	end
	
	def locale
		to(@@identifier)
	end
	
	def to(target)
		text = @raw
		loop do
			separator = text.index("\\")
			return text unless separator
			
			last = text.index("\0")
			return text[separator + 1, last - separator] if text[0, separator] == target[0, separator]
			
			text = text[last + 1, text.length - last - 1]
			while text[0] == " "
				text = text[1, text.length - 1]
			end
		end
	end
	
end

File: python/polystring.py

class Polystring:
	
	identifier = "en_US"
	
	def __init__(self, raw):
		self.raw = raw
	
	def __str__(self):
		return self.locale
	
	def __repr__(self):
		return self.locale
	
	@property
	def locale(self):
		return self.to(Polystring.identifier)
	
	def to(self, target):
		text = self.raw
		while True :
			separator = text.find('\\')
			if separator < 0 :
				return text
			
			end = text.find('\0')
			if text[:separator] == target[:separator] :
				return text[separator + 1: end]
			
			text = text[end + 1:]
			while text[0] == ' ' :
				text = text[1:]
